crates/dslab-mp/src/mc/model_checker.rs

```diff
@@ -99,7 +108,9 @@ impl ModelChecker {
         F: Fn(&mut McSystem),
     {
         let mut total_stats = McStats::default();
-        for state in states {
+        let mut sorted_states = Vec::from_iter(states.into_iter());
+        sorted_states.sort_by_key(|x| x.depth);
+        for state in sorted_states {
```

Comment:
```suggestion
        let mut states = Vec::from_iter(states);
        // sort starting states by increasing depth to produce shorter error traces
        states.sort_by_key(|x| x.depth);
        for state in states {
```

Упростил и добавил комментарий